src/Init/Data/Int/DivMod/Lemmas.lean

```diff
@@ -2330,6 +2330,77 @@ theorem bmod_neg_bmod : bmod (-(bmod x n)) n = bmod (-x) n := by
   apply (bmod_add_cancel_right x).mp
   rw [Int.add_left_neg, ← add_bmod_bmod, Int.add_left_neg]
 
+/- ### sdiv, ediv -/
+
+theorem sdiv_pos_ge_two_lt (x y : Int) (hy' : 2 ≤ y) (hx' : 0 < x):
+    x / y < x := by
+  rw [Int.div_def]
+  unfold Int.ediv
+  obtain ⟨xn, hx⟩ := Int.eq_ofNat_of_zero_le (a := x) (by omega)
+  obtain ⟨yn, hy⟩ := Int.eq_ofNat_of_zero_le (a := y) (by omega)
+  simp only [hx, hy, Int.ofNat_eq_coe, Int.ofNat_lt]
+  apply Nat.div_lt_self (by omega) (by omega)
+
+theorem sdiv_pos_ge_two_ge (x y : Int) (hy' : 2 ≤ y) (hx' : 0 < x):
+    0 ≤ x / y := by
+  rw [Int.div_def]
+  unfold Int.ediv
+  obtain ⟨xn, hx⟩ := Int.eq_ofNat_of_zero_le (a := x) (by omega)
+  obtain ⟨yn, hy⟩ := Int.eq_ofNat_of_zero_le (a := y) (by omega)
+  simp only [hx, hy, Int.ofNat_eq_coe, Int.ofNat_lt]
+  exact Int.ofNat_zero_le (xn / yn)
+
+theorem sdiv_lt_neg_one_le_neg_two_lt (x y : Int) (hy' : y ≤ -2) (hx' : x < -1) :
+    x / y < x.natAbs := by
+  rw [Int.div_def]
+  unfold Int.ediv
+  obtain ⟨xn, hx⟩ := Int.eq_negSucc_of_lt_zero (a := x) (by omega)
+  obtain ⟨yn, hy⟩ := Int.eq_negSucc_of_lt_zero (a := y) (by omega)
+  simp only [hx, hy, Nat.succ_eq_add_one, Int.ofNat_eq_coe, Int.natCast_add, Int.cast_ofNat_Int]
+  norm_cast
+  rw [natAbs_negSucc, Nat.succ_eq_add_one, Nat.add_lt_add_iff_right,
+    Nat.div_lt_iff_lt_mul (x := xn) (k := yn + 1) (y := xn) (by omega),
+    show (xn < xn * (yn + 1)) = (1 * xn < (yn + 1) * xn) by rw [Nat.one_mul, Nat.mul_comm]]
+  apply Nat.mul_lt_mul_of_lt_of_le (a := 1) (b := xn) (c := yn + 1) (d := xn) (by omega) (by omega) (by omega)
+
+theorem sdiv_neg_le_neg_two_ge (x y : Int) (hy' : y ≤ -2) (hx' : x < 0) :
+    0 ≤ x / y := by
+  rw [Int.div_def]
+  unfold Int.ediv
+  obtain ⟨xn, hx⟩ := Int.eq_negSucc_of_lt_zero (a := x) (by omega)
+  obtain ⟨yn, hy⟩ := Int.eq_negSucc_of_lt_zero (a := y) (by omega)
+  simp only [hx, hy, Nat.succ_eq_add_one, ofNat_eq_coe, Int.natCast_add, cast_ofNat_Int]
+  norm_cast
+  simp
+
+theorem sdiv_neg_ge_two_ge (x y : Int) (hy' : 2 ≤ y) (hx' : x < 0) :
```

Comment:
```suggestion
theorem div_le_of_two_le_of_lt_zero (x y : Int) (hy' : 2 ≤ y) (hx' : x < 0) :
```

I wonder what the canonical way of spelling `x < 0` is in the theorem name: do we say `lt_zero`?